src/actions/lint.ts

```diff
@@ -1,128 +1,12 @@
-import { resolve } from 'path';
-import { CLIEngine as Eslint } from 'eslint';
-
-import { getConfigFile } from '../config-file';
+import { LintOptions, lint, report } from '../api';
 import { Action } from './types';
 
-interface LintOptions {
-  config?: string;
-  pattern?: string;
-  typescript?: boolean;
-  javascript?: boolean;
-  includeJsx?: boolean;
-  sass?: boolean;
-  fix?: boolean;
-  spec?: boolean;
-}
-
-interface LintReport {
-  logic?: Eslint.LintReport;
-  style?: {};
-}
-
-const getJsxEtensions = (extensions: string[]): string[] => extensions.map(ext => `${ext}x`);
-const getTestExtensions = (extensions: string[]): string[] => extensions.map(ext => `.spec${ext}`);
-
-const getExtensions = (
-  typescript?: boolean,
-  javascript?: boolean,
-  includeJsx?: boolean,
-): string[] => {
-  let extensions: string[] = [];
-  if (typescript) {
-    extensions.push('.ts');
-  }
-  if (javascript) {
-    extensions.push('.js');
-  }
-  if (includeJsx) {
-    extensions = extensions.concat(getJsxEtensions(extensions));
-  }
-  return extensions;
-};
-
-const hasAnyLintingSet = ({
-  javascript,
-  typescript,
-  sass,
-}: LintOptions): boolean => !!javascript || !!typescript || !!sass;
-
-const setDefaultOptions = (options: LintOptions): LintOptions => ({
-  ...options,
-  pattern: options.pattern || './',
-  javascript: !hasAnyLintingSet(options) || options.javascript,
-  typescript: !hasAnyLintingSet(options) || options.typescript,
-});
-
-const report = ({ logic, style }: LintReport): void => {
-  const projectPath = resolve(process.cwd());
-  let hasErrors = false;
-
-  if (logic) {
-    logic.results.forEach(error => error.messages.forEach((message) => {
-      const fileName = error.filePath.replace(projectPath, '.');
-      const line = `[${message.line}, ${message.column}]`;
-      const rule = `(${message.ruleId})`;
-      // eslint-disable-next-line no-console
-      console.error(`${fileName} ${line}: ${message.message} ${rule}`);
-      hasErrors = true;
-    }));
-  }
-
-  if (style) {
-    // eslint-disable-next-line no-console
-    console.log('SCSS Reporting: Not implemented yet…');
-  }
-
-  if (hasErrors) {
-    process.exit(1);
+const lintAction = (options: LintOptions): void => {
+  if (report(lint(options))) {
+    process.exit(-1);
```

Comment:
Yes. The `-1` exit code is thrown because we don't really care about what exit code is given to the OS. We just want a non-zero exit code so that the process fails.